hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledTaskDescriptor.java

```diff
@@ -90,6 +117,79 @@ void setScheduledFuture(ScheduledFuture<?> future) {
         this.scheduledFuture = future;
     }
 
+    void setResult(Object result) {
+        this.result.set(result);
+    }
+
+    Object getResult()
+            throws ExecutionException, InterruptedException {
+
+        return unpeelFutureOrRoot().get();
```

Comment:
Should check if the result has already been calculated. Currently the result is moved to the other replica only to check the task state.